provider/postgis/postgis.go

```diff
@@ -195,9 +203,144 @@ type hstoreOID struct {
 	hasInit bool
 }
 
+// validateURI validates for minimum requirements for a valid postgresql uri
+func validateURI(u string) error {
+	uri, err := url.Parse(u)
+	if err != nil {
+		return ErrInvalidURI{Err: err}
+	}
+
+	if uri.Scheme != "postgres" && uri.Scheme != "postgresql" {
+		return ErrInvalidURI{
+			Msg: fmt.Sprintf("invalid connection scheme (%v)", uri.Scheme),
+		}
+	}
+
+	if uri.User == nil {
+		return ErrInvalidURI{Msg: "auth credentials missing"}
+	}
+
+	host, port, err := net.SplitHostPort(uri.Host)
+	if err != nil {
+		return ErrInvalidURI{
+			Err: fmt.Errorf("splitting host port error: %w", err),
+		}
+	}
+
+	if host == "" {
+		return ErrInvalidURI{
+			Msg: fmt.Sprintf("address %v:%v: missing host in address", host, port),
+		}
+	}
+
+	if uri.Path == "" {
+		return ErrInvalidURI{Msg: "missing database"}
+	}
+
+	return nil
+}
+
+// BuildURI creates a database URI from config
+func BuildURI(config dict.Dicter) (*url.URL, *url.Values, error) {
```

Comment:
I think we should deprecate the old connection method in favor of using the postgresql URI string. We should do this over 2 versions, so this version we should keep backwards compatibility, but the next one we can remove it. That considered, let's add a TODO code comment here explaining that this function can go away after the cutover is complete.

Reply:
✅